pallets/spin-polkadot/src/lib.rs

```diff
@@ -1,289 +1,195 @@
-	/// A GRANDPA justification for block finality, it includes a commit message
-	/// and an ancestry proof including all headers routing all precommit
-	/// target blocks to the commit target block. Due to the current voting
-	/// strategy the precommit targets should be the same as the commit target,
-	/// since honest voters don't vote past authority set change blocks.
-	///
-	/// See the [original reference](https://github.com/paritytech/polkadot-sdk/blob/polkadot-stable2412-2/substrate/primitives/consensus/grandpa/src/lib.rs#L133)
-	#[derive(CloneNoBound, Encode, Decode, PartialEqNoBound, RuntimeDebugNoBound, TypeInfo)]
-	#[scale_info(skip_type_params(MaxVotesAncestries))]
-	pub struct GrandpaJustification<MaxVotesAncestries: Get<u32>, H: Header> {
-		pub round: u64,
-		// TODO: replace with bounded size data structure
-		pub commit: Commit<H>,
-		pub votes_ancestries: BoundedVec<H, MaxVotesAncestries>,
+	/// Stored configuration for the GRANDPA authority set that signs FastChain finality proofs.
+	#[derive(Clone, Encode, Decode, TypeInfo, PartialEq, Eq)]
+	pub struct AuthoritySetData {
+		pub set_id: SetId,
+		pub authorities: AuthorityList,
 	}
 
-	/// Alive message proof combining `FastchainInherentData` and
-	/// `GrandpaJustification`.
-	///
-	/// Should be sent from a fastchain node to the parachain SPIN pallet via an
-	/// extrinsic call.
-	#[derive(CloneNoBound, Encode, Decode, PartialEqNoBound, RuntimeDebugNoBound, TypeInfo)]
-	#[scale_info(skip_type_params(MaxVotesAncestries))]
-	pub struct AliveMessageProof<MaxVotesAncestries: Get<u32>, H: Header> {
-		pub fastchain_inherent_data: FastchainInherentData,
-		pub grandpa_justification: GrandpaJustification<MaxVotesAncestries, H>,
+	/// Metadata about the best FastChain block accepted on the parachain.
+	#[derive(Clone, Encode, Decode, TypeInfo, PartialEq, Eq, Default)]
+	pub struct FinalizedTarget<Hash> {
+		pub number: u32,
+		pub hash: Hash,
 	}
 
 	#[pallet::config]
 	pub trait Config: frame_system::Config {
 		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
-
-		#[pallet::constant]
-		type TimeoutBlocks: Get<BlockNumberFor<Self>>;
-
-		#[pallet::constant]
-		type CoolDownPeriodBlocks: Get<BlockNumberFor<Self>>;
-
-		#[pallet::constant]
-		type MaxVotesAncestries: Get<u32>;
 	}
 
 	#[pallet::pallet]
 	#[pallet::without_storage_info]
-	pub struct Pallet<T>(_);
+	pub struct Pallet<T>(PhantomData<T>);
 
-	#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, PartialEq)]
-	pub enum SlowchainState<BlockNumber: Clone + PartialEq> {
-		Operational { last_alive_message_block_number: BlockNumber },
-		CoolDown { start_block_number: BlockNumber },
-	}
-
-	/// State of the slowchain: Operational, CoolDown or None
+	/// Current GRANDPA authority set information.
 	#[pallet::storage]
-	pub type State<T: Config> = StorageValue<_, SlowchainState<BlockNumberFor<T>>>;
+	pub type AuthoritySet<T: Config> = StorageValue<_, AuthoritySetData, OptionQuery>;
```

Comment:
Option query is the default and can be omitted, see https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/types/struct.StorageValue.html.

Reply:
True, fixed